service/service.go

```diff
@@ -83,9 +81,8 @@ type serviceConfig struct {
 	Roles       []string `yaml:"roles"`
 }
 
-// New creates a service owner from a set of service instances and options
-// TODO(glib): Something is fishy here... `service.New` returns a service.Owner -_-
-func New(options ...Option) (Owner, error) {
+// NewOwner creates a service owner from a set of module creation functions and options.
+func NewOwner(modules []ModuleCreateFunc, options ...Option) (Owner, error) {
```

Comment:
an even more aggressive thought is to make this function private, and we always encourage people to use `Builder` to build the owner.